src/main/java/ch/unisg/ics/interactions/wot/td/clients/UriTemplate.java

```diff
@@ -0,0 +1,145 @@
+package ch.unisg.ics.interactions.wot.td.clients;
+
+import ch.unisg.ics.interactions.wot.td.schemas.DataSchema;
+
+import java.util.*;
+
+public final class UriTemplate {
```

Comment:
This is not very OO. Why not create a regular object UriTemplate instead of using only static methods?